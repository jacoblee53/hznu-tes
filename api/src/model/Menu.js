import mongoose from 'mongoose'

const Schema = module.Schema

const menuSchema = new mongoose.Schema({
  menuType: {
    type: Number,
  },
  menuContent: [
    {
      url: String,
      path: String,
    },
  ],
})

export default mongoose.model('Menu', menuSchema)
