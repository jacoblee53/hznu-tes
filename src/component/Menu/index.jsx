import React from 'react'
import { Menu, Layout, Icon  } from 'antd'
import { version } from '../../../package.json'
import { NavLink } from 'react-router-dom'

import Logo from '../../constant/image/logo.png'
import './index.less'

const { Sider } = Layout
class DashMenu extends React.Component {
  render() {
    const { collapsed, items } = this.props

    return (
      <Sider
        trigger={null}
        collapsible
        collapsed={collapsed}
        width={256}
        theme='dark'
      >
        <div className='logo'>
          <a href='/'>
            <img src={Logo} />
            <h1>学科竞赛评价系统</h1>
          </a>
        </div>
        <Menu
          theme='dark'
          mode='inline'
          className='menu'
        >
          {items.map(item => (
            <MenuItem key={item.path}>
              <NavLink to={item.path}>
                <Icon type={item.icon} />
                <span>{item.title}</span>
              </NavLink>
            </MenuItem>
          ))}
        </Menu>
      </Sider>
    )
  }
}

export default DashMenu