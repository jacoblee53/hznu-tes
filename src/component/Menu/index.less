.logo {
  height: 50px;
  line-height: 50px;
  position: relative;
  padding-left: 24px;
  overflow: hidden;
  background: #001529;
  transition: all .3s;

  a {
    background-color: transparent;
    color: #1890ff;
    cursor: pointer;
    outline: none;
    text-decoration: none;
    transition: color .3s;
  }

  img {
    display: inline-block;
    height: 32px;
    vertical-align: middle;
  }

  h1 {
    color: #fff;
    display: inline-block;
    font-size: 16px;
    font-weight: normal;
    margin: 0 0 0 12px;
    vertical-align: middle;
  }
}