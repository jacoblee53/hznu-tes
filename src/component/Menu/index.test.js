import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import injects from '../../inject/index'
import Menu from './index'

let wrappedComponent = <Menu {...injects} />

let wrapper

describe('src/app/my-task/component/index.test.js', () => {
  it('should render correctly', () => {
    wrapper = mount(wrappedComponent)
    expect(wrapper.html()).toBeTruthy()
  })

  it('should matchSnapshot', () => {
    const tree = renderer.create(wrappedComponent)
    expect(tree).toMatchSnapshot()
  })
})
