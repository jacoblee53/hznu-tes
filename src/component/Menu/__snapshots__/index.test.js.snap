// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/my-task/component/index.test.js should matchSnapshot 1`] = `
<aside
  className="ant-layout-sider ant-layout-sider-dark"
  style={
    Object {
      "flex": "0 0 256px",
      "maxWidth": "256px",
      "minWidth": "256px",
      "width": "256px",
    }
  }
>
  <div
    className="ant-layout-sider-children"
  >
    <div
      className="logo"
    >
      <a
        href="/"
      >
        <img
          src="logo.png"
        />
        <h1>
          学科竞赛评价系统
        </h1>
      </a>
    </div>
    <ul
      className="ant-menu menu ant-menu-dark ant-menu-root ant-menu-inline"
      onMouseEnter={[Function]}
      onTransitionEnd={[Function]}
      role="menu"
      style={Object {}}
    >
      <div
        className="version"
      >
        v 
        1.0.0
      </div>
    </ul>
  </div>
</aside>
`;
